src/scheduler/Analysis.py

```diff
@@ -146,13 +149,70 @@ def scheduler(self):
                 self.process_next_analysis(task)
 
-    def process_next_analysis(self, fw_object):
+    def process_next_analysis(self, fw_object: FileObject):
         self.pre_analysis(fw_object)
         analysis_to_do = fw_object.scheduled_analysis.pop()
         if analysis_to_do not in self.analysis_plugins:
             logging.error('Plugin \'{}\' not available'.format(analysis_to_do))
+            self.check_further_process_or_complete(fw_object)
         else:
+            self._start_or_skip_analysis(analysis_to_do, fw_object)
+
+    def _start_or_skip_analysis(self, analysis_to_do, fw_object):
+        if analysis_to_do in MANDATORY_PLUGINS or self._next_analysis_is_not_blacklisted(analysis_to_do, fw_object):
             self.analysis_plugins[analysis_to_do].add_job(fw_object)
+        else:
+            logging.debug('skipping analysis "{}" for {} (blacklisted file type)'.format(analysis_to_do, fw_object.get_uid()))
+            fw_object.processed_analysis[analysis_to_do] = self._get_skipped_analysis_result(analysis_to_do)
+            self.check_further_process_or_complete(fw_object)
+
+    def _get_skipped_analysis_result(self, analysis_to_do):
+        return {
+            'skipped': 'blacklisted file type',
+            'summary': [],
+            'analysis_date': time(),
+            'plugin_version': self.analysis_plugins[analysis_to_do].VERSION
+        }
+
+    # ---- blacklist and whitelist ----
+
+    def _next_analysis_is_not_blacklisted(self, next_analysis, fw_object: FileObject):
+        blacklist, whitelist = self._get_blacklist_and_whitelist(next_analysis)
+        if not (blacklist or whitelist):
+            return True
+        if blacklist and whitelist:
+            logging.error('{}Configuration of plugin "{}" erroneous{}: found blacklist and whitelist. Ignoring blacklist.'.format(
+                bcolors.FAIL, next_analysis, bcolors.ENDC))
+
+        try:
+            file_type = fw_object.processed_analysis['file_type']['mime'].lower()
+        except KeyError:  # FIXME file_type analysis is missing (probably due to problem with analysis caching) -> re-schedule
+            fw_object.scheduled_analysis.extend([next_analysis, 'file_type'])
+            fw_object.analysis_dependency.add('file_type')
+            return False
+
+        if whitelist:
+            return substring_is_in_list(file_type, whitelist)
+        return not substring_is_in_list(file_type, blacklist)
+
+    def _get_blacklist_and_whitelist(self, next_analysis):
+        blacklist, whitelist = self._get_blacklist_and_whitelist_from_config(next_analysis)
+        if not (blacklist or whitelist):
+            blacklist, whitelist = self._get_blacklist_and_whitelist_from_plugin(next_analysis)
+        return blacklist, whitelist
+
+    def _get_blacklist_and_whitelist_from_config(self, analysis_plugin: str) -> Tuple[List, List]:
+        blacklist = self.config.get(analysis_plugin, 'mime_blacklist', fallback='').split(', ')
```

Comment:
There should be a helper function parsing the list
"a,b" should be as valid as "a, b"